mobile/views/UserOnboarding/ExploreCommunities/CommunityProfileCard/index.js

```diff
@@ -0,0 +1,78 @@
+// @flow
+import * as React from 'react';
+import { View } from 'react-native';
+import { type GetCommunityType } from '../../../../../shared/graphql/queries/community/getCommunity';
+import {
+  CommunityCardWrapper,
+  CommunityProfilePhoto,
+  CommunityProfileName,
+  CommunityProfileMemberCount,
+  CommunityProfileDescription,
+  CommunityProfileButtonContainer,
+  CommunityProfileHeader,
+} from './style';
+import LoadingCard from './LoadingCard';
+import { Button } from '../../../../components/Button';
+
+type Props = {
+  community: GetCommunityType,
+  joinedCommunities: Array<?string>,
+  onJoinCommunity: Function,
+  onLeaveCommunity: Function,
+};
+
+class CommunityCard extends React.Component<Props> {
+  join = () => {
+    const { community } = this.props;
+    return this.props.onJoinCommunity(community.id);
+  };
+
+  leave = () => {
+    const { community } = this.props;
+    return this.props.onLeaveCommunity(community.id);
+  };
+
+  render() {
+    const { community, joinedCommunities } = this.props;
+    const hasJoined = joinedCommunities.indexOf(community.id) >= 0;
+    return (
+      <CommunityCardWrapper>
+        <View>
+          <CommunityProfileHeader>
+            <CommunityProfilePhoto source={{ uri: community.profilePhoto }} />
+            <CommunityProfileButtonContainer>
+              {hasJoined ? (
+                <Button
+                  onPress={this.leave}
+                  icon={'checkmark'}
+                  color={props => props.theme.text.default}
+                />
+              ) : (
+                <Button
+                  onPress={this.join}
+                  title={'Join'}
```

Comment:
nit: the join button on the community cards feels a bit out of place/squashed in. no real ideas how to solve that though, putting them full-width at the bottom would def be too much.

Reply:
Yeah, I spent a while on this - not sure exactly what to do yet to improve this :/